plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -234,6 +254,137 @@ public void onFailure(Exception e) {
         });
     }
 
+    private void checkAgentBeforeDeleteModel(String modelId, ActionListener<Boolean> actionListener) {
+        // check whether agent are using them
+        SearchRequest searchAgentRequest = agentModelsSearcher.constructQueryRequest(modelId);
+        client.search(searchAgentRequest, ActionListener.wrap(searchResponse -> {
+            SearchHit[] searchHits = searchResponse.getHits().getHits();
+            if (searchHits.length == 0) {
+                actionListener.onResponse(true);
+            } else {
+                List<String> relatedAgents = new ArrayList<>();
+                for (SearchHit hit : searchHits) {
+                    relatedAgents.add(hit.getId());
+                }
+                actionListener
+                    .onFailure(
+                        new OpenSearchStatusException(
+                            searchHits.length
+                                + " agents are still using this model, please delete or update the agents first: "
+                                + Arrays.toString(relatedAgents.toArray(new String[0])),
+                            RestStatus.CONFLICT
+                        )
+                    );
+            }
+
+        }, e -> {
+            if (e instanceof IndexNotFoundException) {
+                actionListener.onResponse(true);
+                return;
+            }
+            log.error("Failed to delete ML Model: " + modelId, e);
+            actionListener.onFailure(e);
+
+        }));
+    }
+
+    private void checkIngestPipelineBeforeDeleteModel(String modelId, ActionListener<Boolean> actionListener) {
+        GetPipelineRequest getPipelineRequest = new GetPipelineRequest();
+        client.execute(GetPipelineAction.INSTANCE, getPipelineRequest, ActionListener.wrap(ingestPipelineResponse -> {
+            if (!isPipelineContainsModel(
+                ingestPipelineResponse.pipelines(),
+                modelId,
+                org.opensearch.ingest.PipelineConfiguration::getConfigAsMap
```

Comment:
Since the return type of ingestPipelineResponse.pipelines() is List<PipelineConfiguration>, why not simply call pipelineConfiguration.getConfigAsMap()?

Reply:
No. Ingestion and search pipeline are two different types.